zng/docs/spec.md

```diff
@@ -9,10 +13,13 @@
 > * the `bytes` type is not yet implemented,
 > * the `enum` type is not yet implemented,
 > * only streams of `record` types (which may consist of any combination of
->   other implemented types) may currently be expressed in value messages.
+>   other implemented types) are supported by zq even though a stream of
+>   any types may currently be expressed in value messages.
 >
-> Also, we are contemplating reducing the number of [primitive types](#5-primitive-types), e.g.,
-> the number of variations in integer types.
+> TBD: should we add maps? It would be a complex type comprising
+> a list of key/values where the key type and value type is defined in the
+> map typedef.  The contrarian view is that maps are easily represented
```

Comment:
Note that the contrarian view does not preclude having maps in ZQL at some point in the future. There could be a operator `map(list)` that turns a list of `record[key:keyType,value:valueType]`  into a map that can then be operated in by map-savvy procs and reducers in ZQL. Or for more flexibility, maybe `map()` takes the key/value field names, so you could say `map(list, key, value)`. Maps would have to be converted back into a list representation before exiting the flowgraph. 

There might also be a way to create maps from a stream of a records, for example `count() by _path | mapStream(_path, count)` would transform the stream of records into a histogram-like map of `string->int` which would be more suitable e.g. for statistical operations than separate records.

Anyway, spitballing here and not suggesting we need to do this... but it shows how the space of co-designing a data format and data language is novel broad. We're just scratching the outer surface here. 😎